c/dataservice.c

```diff
@@ -109,6 +109,18 @@ static void *lookupDLLEntryPoint(char *libraryName, char *functionName){
     ;
   void *library = dlopen(libraryName,flags);
   if (library){
+    FileInfo info;
+    int status = 0;
+    int returnCode = 0;
+    int reasonCode = 0;
+    status = fileInfo(libraryName, &info, &returnCode, &reasonCode);
+    if (status == 0) {
+      if (!(info.attributeFlags & 0x2)) {
+        printf("WARNING: Plugin: %s does not have the Program Control attribute this may cause unexpected errors\n", libraryName);
```

Comment:
Should probably just use zowelog. We have a logging component id for dataservice.c. You might need to configure it at server startup.